tcms/kiwi_auth/tests/test_views.py

```diff
@@ -262,3 +278,49 @@ def test_confirm(self):
         self.assertTrue(user.is_active)
         activate_key_deleted = not UserActivationKey.objects.filter(user=user).exists()
         self.assertTrue(activate_key_deleted)
+
+
+class TestLoginViewWithCustomTemplate(TestCase):
+    """Test for login view with custom template"""
+
+    def test_get_template_names(self):
+        try:
+            response = self.client.get(reverse("tcms-login"))
+            self.assertIsNotNone(response.template_name)
+            self.assertEqual(
+                response.template_name,
+                ["registration/custom_login.html", "registration/login.html"],
+            )
+        except TemplateDoesNotExist:
+            self.fail("Failed to load template")
+
+
+class TestPasswordResetView(TestCase):
+    """Test for password reset view"""
+
+    def test_form_class(self):
+        response = self.client.get(reverse("tcms-password_reset"))
+        self.assertIsInstance(response.context["form"], forms.PasswordResetForm)
+
+
+class TestProfileView(TestCase):
+    """Test user profile view"""
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.tester = UserFactory()
+        cls.tester.set_password("password")
+        cls.tester.save()
+
+    def test_view_if_user_is_valid(self):
+        self.client.login(  # nosec:B106:hardcoded_password_funcarg
+            username=self.tester.username, password="password"
+        )
+        response = self.client.get(
+            reverse("tcms-profile", args=[self.tester.username]), follow=True
+        )
+        self.assertRedirects(response, "/admin/auth/user/2/change/", status_code=302)
+
+    def test_view_if_user_is_invalid(self):
+        response = self.client.get(reverse("tcms-profile", args=["kiwi-tester"]))
+        self.assertEqual(response.status_code, 404)
```

Comment:
this isn't so you can merge it with test_profiles.py and adjust the existing test class to reuse LoggedInTestCase